main/distributed-programming.md

```diff
@@ -68,22 +68,17 @@ Agoric "deploy scripts" and smart contract code run in an immutable realm with c
 
 ### harden
 
-`harden` is a function provided by the [`@agoric/harden` package](https://www.npmjs.com/package/@agoric/harden).
-
-In a nutshell, `harden` creates objects that can't be changed. In practice,
-`harden` performs the deep version of
-[`Object.freeze`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze). Freeze
-only freezes the object's own properties, so harden is necessary in order to
-ensure that no further changes can be made. It is useful in distributed systems
-to distinguish objects with mutable state from objects that are
-unchanging. Objects that can change over time are extremely useful, but in order
-to interact with them, you have to send the message to where they are. If an
-object is guaranteed not to change, then copies of it can be widely shared, and
-interacted with directly.
-
-Many services provided by Agoric require their parameters to be hardened so they
-can interact with them locally. This makes it possible to provide reliable
-service without waiting for further round trips to collect information.
+`harden` is a function provided by the
+[`@agoric/harden` package](https://www.npmjs.com/package/@agoric/harden).  It ensures
+that external callers of hardened objects can only interact with them through
+functions present in their API.  `harden` does a transitive version of
+[`Object.freeze`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze),
+which only locks up an object's own properties.
+
+CapTP, our comms layer for passing references to distributed objects, requires that
+all objects that will be transferred to other trust domains must have their API
```

Comment:
"trust domains" is not accurate. There's a trust boundary between the zoe contract facet and the contract in the same vat, but there is no captp there, and nothing gets marshaled across that boundary. Nonetheless, because it is a security boundary, each side *should* harden the objects they expose to the other.

As for the captp/marshall-imposed requirement, that's only at vat boundaries.

Reply:
I changed it to say 
```
All objects that will be transferred across a trust boundary must have their API surface frozen
```
and mention at the end that CapTP enforces the rule across vat boundaries.